Functional.Primitives.Extensions/ResultExtensions.cs

```diff
@@ -129,5 +129,32 @@ public static class ResultExtensions
 
 		public static async Task<Result<TSuccess, TFailure>> FailureIfNone<TSuccess, TFailure>(this Task<Result<Option<TSuccess>, TFailure>> result, Func<TFailure> failureFactory)
 			=> (await result).FailureIfNone(failureFactory);
+
+		public static Result<TResult, TFailure> TrySelect<TSuccess, TResult, TFailure>(this Result<TSuccess, TFailure> result, Func<TSuccess, TResult> successFactory, Func<Exception, TFailure> failureFactory)
+		{
+			if (successFactory == null)
+				throw new ArgumentNullException(nameof(successFactory));
+
+			if(failureFactory == null)
+				throw new ArgumentNullException(nameof(failureFactory));
+
+			try
+			{
```

Comment:
Sorry to knitpick, but should we use a Bind here for consistency with the async endpoint?

Reply:
Absolutely!